packages/test/test-end-to-end-tests/src/test/blobs.spec.ts

```diff
@@ -432,3 +434,240 @@ describeNoCompat("blobs", (getTestObjectProvider) => {
+            // we don't expose the storageId for the blob's uploaded in detached container.
+            const blobHandle2 = defaultDataStore2._root.get<IFluidHandle<ArrayBufferLike>>("blob");
+            assert.strictEqual(blobHandle.absolutePath, blobHandle2?.absolutePath,
+                    "The blob handle has a different path in remote container.");
+
+            const s1 = await getUnreferencedNodeStates();
+            assert(s1.get(blobHandle.absolutePath) === "referenced", "blob should be referenced");
+
+            // Remove the blob's handle and verify its marked as unreferenced.
+            defaultDataStore._root.delete("blob");
+            const s2 = await getUnreferencedNodeStates();
+            assert(s2.get(blobHandle.absolutePath) === "unreferenced", "blob should be unreferenced");
+
+            // Add the blob's handle in second container. If deleteUnreferencedContent is true, the blob's node would
+            // have been deleted from the GC state. Else, it would be referenced.
+            defaultDataStore2._root.set("blobContainer2", blobHandle2);
+            const s3 = await getUnreferencedNodeStates();
+            if (deleteUnreferencedContent) {
+                assert(s3.get(blobHandle.absolutePath) === undefined, "blob should not have a GC entry");
+            } else {
+                assert(s3.get(blobHandle.absolutePath) === "referenced", "blob should be re-referenced");
+            }
+        });
+
+        it("collects blobs uploaded in detached and de-duped in attached container", async () => {
+            // Upload an attachment blob. We should get a handle with a localId for the blob. Mark it referenced by
+            // storing its handle in a DDS.
+            const blobContents = "Blob contents";
+            const localHandle = await defaultDataStore._context.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+            defaultDataStore._root.set("blob", localHandle);
+
+            // Attach the container after the blob is uploaded.
+            await container.attach(provider.driver.createCreateNewRequest(provider.documentId));
+
+            // Upload the same blob. This will get de-duped and we will get back a handle with the stoageId instead of
+            // the localId that we got when uploading in detached container.
+            const storageHandle = await defaultDataStore._context.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+
+            // Validate that storing the localId handle makes both the localId and storageId nodes as referenced since
+            // localId is simply an alias to the storageId.
+            const s1 = await getUnreferencedNodeStates();
+            assert(s1.get(localHandle.absolutePath) === "referenced", "local id blob should be referenced");
+            assert(s1.get(storageHandle.absolutePath) === "referenced", "storage id blob should also be referenced");
```

Comment:
Would this allow a blob be doubly referenced in the GC graph?

Reply:
By doubly referenced, do you mean two nodes having an outbound route to this node? If so, yes, and this is always supported where a handle to a node can be stored in multiple DDSs.
In this particular case, we have a localId and storageId for the same underlying blob. Storing a handle to any one (or both) of them will the blob referenced.